maintainers_annotation/maintainers_annotation.go

```diff
@@ -0,0 +1,190 @@
+	AnnotationRemediated:    "The dangerous files or code snippets are necessary but remediations were already applied.",
+	AnnotationNotApplicable: "The check or probe is not applicable in this case.",
+	AnnotationNotSupported:  "The check or probe is fulfilled but in a way that is not supported by Scorecard.",
+	AnnotationNotDetected:   "The check or probe is fulfilled but in a way that is supported by Scorecard but it was not detected.",
+}
+
+// Annotation groups the annotation reason and, in the future, the related probe.
+// If there is a probe, the reason applies to the probe.
+// If there is not a probe, the reason applies to the check or group of checks in
+// the exemption.
+type Annotation struct {
+	AnnotationReason AnnotationReason `yaml:"annotation"`
+}
+
+// Exemption defines a group of checks that are being annotated for various reasons.
+type Exemption struct {
+	Checks      []string     `yaml:"checks"`
+	Annotations []Annotation `yaml:"annotations"`
+}
+
+// MaintainersAnnotation is a group of exemptions.
+type MaintainersAnnotation struct {
+	Exemptions []Exemption `yaml:"exemptions"`
+}
+
+// parseScorecardYmlFile takes the scorecard.yml file content and returns a `MaintainersAnnotation`.
+func parseScorecardYmlFile(ma *MaintainersAnnotation, content []byte) error {
+	unmarshalErr := yaml.Unmarshal(content, ma)
+	if unmarshalErr != nil {
+		return sce.WithMessage(sce.ErrScorecardInternal, unmarshalErr.Error())
+	}
+
+	return nil
+}
+
+// validateAndParseScorecardYmlFile takes the scorecard.yml files found, validates if its the correct
+// configuration file and parse its configurations.
+func validateAndParseScorecardYmlFile(path string, content []byte, args ...interface{}) (bool, error) {
+	// Verify if received the necessary parameters to save the scorecard YML configurations
+	if len(args) != 1 {
+		return false, fmt.Errorf(
+			"validateAndParseScorecardYmlFile requires exactly one argument to save maintainers annotation data")
+	}
+	ma, ok := args[0].(*MaintainersAnnotation)
+	if !ok {
+		return false, fmt.Errorf(
+			"validateAndParseScorecardYmlFile requires argument of type *MaintainersAnnotation")
+	}
+
+	// Exclude scorecard GitHub Action file
+	if strings.Contains(path, ".github") {
+		return true, nil
+	}
+
+	// Keep looking for files if content is empty
+	if len(content) == 0 {
+		return true, nil
+	}
+
+	// Found scorecard.yml file, parse file
+	err := parseScorecardYmlFile(ma, content)
+	if err != nil {
+		return false, err
+	}
+
+	return false, nil
+}
+
+// readScorecardYmlFromRepo reads the scorecard.yml file from the repo and returns its configurations.
+func readScorecardYmlFromRepo(repoClient clients.RepoClient) (MaintainersAnnotation, error) {
+	ma := MaintainersAnnotation{}
+	err := fileparser.OnMatchingFileContentDo(repoClient, fileparser.PathMatcher{
+		Pattern:       "scorecard.yml",
+		CaseSensitive: false,
+	}, validateAndParseScorecardYmlFile, &ma)
```

Comment:
Simpler to use `GetFileContent`? 

This approach would match any file named "scorecard.yml", which could be the Scorecard Action workflow file for example. Which I see you have a check for elsewhere, but if we have specific spots to look, might be easier to just look there.

Reply:
Done.

Reply:
Thanks! I was looking for such a function but I could not find it. Works perfectly.